### PR TITLE
[ROCm] Switching to rocblas_gemm_ex for MFMA-enabled architectures

### PR DESCRIPTION
**Pull request by deven-amd**
Filing the PR to upstream the same submitted by @ekuznetsov139 in the rocmfork.

This PR adds two pieces of functionality.
1. `GetMFMASupport` routine (which is ROCM specific) to determine whether the underlying AMDGPU has support for mfma instructions
2. Call the rocblas GEMM API that leverages the mfma instructions (when available and applicable)

-----------------------------

/cc @chsigg @cheshire

### CODE REVIEW

**Comment by deven-amd** [comment]
@googlebot I consent.

**Comment by gbaned** [comment]
@deven-amd Can you please sign CLA. Thanks!

**Comment by ekuznetsov139** [comment]
@googlebot I consent

**Comment by deven-amd** [comment]
@chsigg @cheshire gentle ping

**Comment by deven-amd** [comment]
@cheshire @chsigg gentle ping

**Comment by deven-amd** [comment]
@cheshire @chsigg gentle ping

**Comment by deven-amd** [comment]
@cheshire @chsigg gentle ping

File: tensorflow/stream_executor/rocm/rocm_blas.cc [hunk 1]
```diff
@@ -1617,16 +1618,35 @@ bool ROCMBlas::DoBlasGemm(Stream *stream, blas::Transpose transa,
                       "precondition violation";
     }
   }
-  const Eigen::half alpha_half(alpha);
-  const Eigen::half beta_half(beta);
-  return DoBlasInternal(
+  bool hasXDLOPS = false;
+  auto status = GpuDriver::GetMFMASupport(hasXDLOPS);
```
Comment (cheshire) [inline tensorflow/stream_executor/rocm/rocm_blas.cc:1622]: Explicit type preferred + is it actually checked?

Comment (deven-amd) [reply tensorflow/stream_executor/rocm/rocm_blas.cc]: @cheshire pushed out a commit to make the type explicit and check the return status.

Please re-review.

thanks

File: tensorflow/stream_executor/gpu/gpu_driver.h [hunk 2]
```diff
@@ -474,6 +474,11 @@ class GpuDriver {
   static port::Status GetGpuGCNArchName(GpuDeviceHandle device,
                                         std::string* gcnArchName);
 
+#if TENSORFLOW_USE_ROCM
+  // tests the current device for MFMA insn support (ROCm only)
+  static port::Status GetMFMASupport(bool& support);
```
Comment (cheshire) [inline tensorflow/stream_executor/gpu/gpu_driver.h:479]: Could we use StatusOr<bool> instead?

**Review by cheshire — APPROVED** [review]

File: tensorflow/stream_executor/gpu/gpu_driver.h (hunk 2 above)
Comment (deven-amd) [reply tensorflow/stream_executor/gpu/gpu_driver.h]: done.

please re-review re-approve.

Comment (deven-amd) [reply tensorflow/stream_executor/gpu/gpu_driver.h]: @cheshire gentle ping

Comment (deven-amd) [reply tensorflow/stream_executor/gpu/gpu_driver.h]: @cheshire gentle ping

**Review by cheshire — APPROVED** [review]

**Pull request merged**